File: mifos-passcode/src/main/java/com/mifos/mobile/passcode/utils/EncryptionUtil.java
package com.mifos.mobile.passcode.utils;

import android.util.Log;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.IntDef;

/**
 * Created by dilpreet on 11/7/17.
 */

public class EncryptionUtil {

    public static final int DEFAULT = 1;
    public static final int MOBILE_BANKING = 2;
    public static final int ANDROID_CLIENT = 3;
    public static final int FINERACT_CN = 4;

    @IntDef({DEFAULT, MOBILE_BANKING, ANDROID_CLIENT, FINERACT_CN})
    @Retention(RetentionPolicy.SOURCE)
    public @interface TYPE { };

    static {
        try {
            System.loadLibrary("encryption");
        } catch (UnsatisfiedLinkError e) {
            Log.e("LoadJniLib", "Error: Could not load native library: " + e.getMessage());
        }
    }

    private static final native String getPassCodeHash(String passcode);

    public static String getDefaultHash(String passCode) {
        return getPassCodeHash(passCode);
    }

    public static String getMobileBankingHash(String passCode) {
        return getPassCodeHash(passCode);
    }

    public static String getAndroidClientHash(String passCode) {
        return getPassCodeHash(passCode);
    }

    public static String getFineractCNHash(String passCode) {
        return getPassCodeHash(passCode);
    }
}

File: mifos-passcode/src/main/java/com/mifos/mobile/passcode/utils/EncryptionUtil.kt
package com.mifos.mobile.passcode.utils

import android.util.Log
import androidx.annotation.IntDef

object EncryptionUtil {

    const val DEFAULT = 1
    const val MOBILE_BANKING = 2
    const val ANDROID_CLIENT = 3
    const val FINERACT_CN = 4


    @IntDef(DEFAULT, MOBILE_BANKING, ANDROID_CLIENT, FINERACT_CN)
    @Retention(AnnotationRetention.SOURCE)
    annotation class TYPE

    init {
        try {
            System.loadLibrary("encryption")
        } catch (e: UnsatisfiedLinkError) {
            Log.e("LoadJniLib", "Error: Could not load native library: ${e.message}")
        }
    }

    external fun getPassCodeHash(passcode: String): String

    fun getDefaultHash(passCode: String): String {
        return getPassCodeHash(passCode)
    }

    fun getMobileBankingHash(passCode: String): String {
        return getPassCodeHash(passCode)
    }

    fun getAndroidClientHash(passCode: String): String {
        return getPassCodeHash(passCode)
    }

    fun getFineractCNHash(passCode: String): String {
        return getPassCodeHash(passCode)
    }
}
